source/include/core_http_client_private.h

```diff
@@ -213,12 +213,12 @@ typedef enum HTTPParsingState_t
  */
 typedef struct findHeaderContext
 {
-    const char * pField;     /**< The field that is being searched for. */
-    size_t fieldLen;         /**< The length of pField. */
-    const char ** pValueLoc; /**< The location of the value found in the buffer. */
-    size_t * pValueLen;      /**< the length of the value found. */
-    uint8_t fieldFound;      /**< Indicates that the header field was found during parsing. */
-    uint8_t valueFound;      /**< Indicates that the header value was found during parsing. */
+    const unsigned char * pField; /**< The field that is being searched for. */
```

Comment:
just so later reviewers are aware, this is done since a function that was made for misra compliance that takes an unsigned char * is passed this field, causing a build warning that turns into an error on the CSDK demo pipeline in the CI with warnings being treated as errors.

Reply:
we didn't choose to just change the function signature because then it causes a MISRA violation of passing a signed char to a ctype function.

Reply:
Perhaps caseInsensitiveStringCmp should take ``const char *`` arguments then?

Reply:
The reason we had done it this way is because passing in a char to toupper(), or any function in <ctype.h> is a MISRA 21.13 rule violation, so in order to swap caseInsensitiveStringCmp() to take in an unsigned char would require us to cast each character being passed into toupper().

Reply:
It seems quite odd to use an int for comparison here.